### DISCUSSION TITLE
Exposing metrics for Prometheus from Kamon dashboard.

### DISCUSSION THREAD

**Discussion by GoutamVerma** [ACINQ/eclair#2250] [Developers]
Hello Everyone,

I have figured out the best way to get endpoint metrics.
Kamon is already exposing metrics from the eclair node. The idea is to 
bridge the kamon dashboard and Prometheus. 

In order to implement that thing in our codebase, we have to use 
Gradle dependency. But before doing anything I need to see the Kamon dashboard 
for the eclair node at my end.  I need your little help in seeing Kamon's dashboard 

official doc: https://kamon.io/docs/latest/reporters/prometheus/

After seeing the Kamon dashboard we will be soo closer to generating PRs for it

**Comment by t-bast** [comment]
> we have to use Gradle dependency

Note that we use maven in eclair, not gradle, but it will work mostly the same.

> But before doing anything I need to see the Kamon dashboard for the eclair node at my end.

I don't understand why you need that? We cannot give you access to our private Kamon dashboard.
But that doesn't prevent you in any way from adding the prometheus metrics exporter to eclair in a first PR.
That should be your first step.

You can then create your own grafana dashboards based on the eclair metrics you will receive in Prometheus.
Once you reach that point, spend some time looking at all the metrics you receive, and try creating dashboards on your own based on what seems to make sense.

**Reply by GoutamVerma** [reply]
Okay, we will go for maven!

Actually, instead of making exporter for Prometheus separate, I was planning to bridge it with the Kamon dashboard.
But no issues I will create a separate once generate the first PR for it.

Can I create an issue from this discussion?

**Reply by t-bast** [reply]
> I was planning to bridge it with the Kamon dashboard.

I don't understand what that means. The first step you need to do is to add the ability for eclair to export its metrics to a prometheus endpoint, thanks to the kamon libraries. This is the first PR you should make to eclair.

Once that PR is complete, we will be able to create dashboards, but that first step is completely independent (and necessary).

> Can I create an issue from this discussion?

Why do you need an issue? You can just start working on the code and open a PR once the code is ready.

**Reply by GoutamVerma** [reply]
> but that first step is completely independent (and necessary).

Now the thing is clear will create independent exporters in maven

> Why do you need an issue?

I wanted to make PRs against that issue just for a better understanding but no issue once the code is ready will generate PR

**Comment by GoutamVerma** [comment]
Hello Mentor,

I have encountered with few things while developing an exporter for the eclair node.
Prometheus majorly accepts three types of exporter( HAProxy, MySql, and node exporter).
In our case node exporter is the best option for getting endpoint metrics, cause it is a mix of these with complexity varying by module.

So I am moving further with node exporter type!!
